packages/fast-components-react-base/src/listbox/listbox.tsx

```diff
@@ -0,0 +1,284 @@
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { ListboxClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    ListboxHandledProps,
+    ListboxProps,
+    ListboxUnhandledProps,
+} from "./listbox.props";
+import * as React from "react";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+import { get, inRange, invert } from "lodash-es";
+import { canUseDOM } from "exenv-es6";
+
+export interface ListboxState {
+    /**
+     * The index of the focusable child
+     */
+    focusIndex: number;
+    focussedItemId: string;
+}
+
+export interface TypeAheadDataItem {
+    index: number;
+    compareString: string;
+}
+
+class Listbox extends Foundation<
+    ListboxHandledProps,
+    ListboxUnhandledProps,
+    ListboxState
+> {
+    public static displayName: string = "Listbox";
+
+    public static defaultProps: Partial<ListboxProps> = {
+        multiselectible: false,
+    };
+
+    protected handledProps: HandledProps<ListboxHandledProps> = {
+        children: void 0,
+        managedClasses: void 0,
+        typeAheadPropertyKey: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    private typeAheadString: string = "";
+    private typeAheadTimer: any;
+    private typeAheadData: TypeAheadDataItem[] = [];
+
+    constructor(props: ListboxProps) {
+        super(props);
+
+        this.state = {
+            focusIndex: -1,
+            focussedItemId: "",
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                {...this.unhandledProps()}
+                ref={this.rootElement}
```

Comment:
I think we should consider adding a first-class prop for aria-labelled by. Per the spec:

> If the listbox is not part of another widget, then it has a visible label referenced by aria-labelledby on the element with role listbox.

Reply:
Added that.